lazy-either.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const LazyEither = require('lazy-either');
+const R = require('ramda');
+const S = require('sanctuary');
+
+const lift = LazyEither.lift;
+
+// join :: String -> String -> String
+const join = R.curryN(2, path.join);
+
+// readFile :: String -> String -> LazyEither (Either e String)
+const readFile = R.curry((encoding, filename) =>
+  new LazyEither(resolve => {
+    fs.readFile(filename, {encoding: encoding}, (err, data) =>
+        resolve(err ? S.Left(err) : S.Right(data.toString())));
+  }));
+
+// readFiles :: [String] -> LazyEither (Either e String)
+const readFiles = files => R.traverse(LazyEither.of, readFile('utf8'), files);
+
+// concatFiles :: String -> LazyEither (Either e String)
+const concatFiles = dir =>
+  R.pipeK( readFile('utf8')
+         , lift(S.lines)
+         , lift(R.map(join(dir)))
+         , readFiles
+         , lift(R.join(''))
+         ) (LazyEither.of(join(dir, 'index.txt')));
+
+const main = () => {
+  concatFiles(process.argv[2]).value(data => {
+    if (data.isRight)
+      process.stdout.write(data.value);
+    else
+      process.stderr.write(String(data.value) + '\n');
+    process.exit(data.isRight ? 0 : 1);
```

Comment:
I'd prefer not to branch in two places. Please use the same approach as in the other implementations so readers can focus on the *meaningful* differences between the various implementations.